### ISSUE TITLE
npm install semantic-ui-react error

### ISSUE THREAD

**Issue by block0xhash** [Semantic-Org/Semantic-UI-React#4129]
good day

I am geeting error below trying when I run:
npm install semantic-ui-react

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: client-app@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from semantic-ui-react@2.0.1
npm ERR! node_modules/semantic-ui-react
npm ERR!   semantic-ui-react@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!

**Comment by brianespinosa** [comment]
@bothwellw what is your application toolchain? Also, please share what your package.json file looks like. I am a `yarn` user but it seems like you may not have `react` declared as a dependency in your application. This is just my guess without seeing your package.json and knowing your toolchain.

**Comment by block0xhash** [comment]
{
  "name": "revents",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.6",
    "@testing-library/react": "^11.2.2",
    "@testing-library/user-event": "^12.2.2",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-scripts": "4.0.1",
    "web-vitals": "^0.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}


`
c:\>
c:\>node --version
v15.3.0

c:\>npm --version
7.0.14
`

**Comment by james-work-account** [comment]
@brianespinosa I'm having the same problem as @bothwellw, from using `create-react-app` to create a boilerplate react app and then installing semantic-ui-react with `npm install semantic-ui-react semantic-ui-css`. NPM version 7.1.10, Node version 15.2.0.

**Comment by james-work-account** [comment]
Pushing my React version back to this* makes the error go away:

```
"dependencies": {
  "react": "^16.13.1",
  "react-dom": "^16.13.1",
  "react-scripts": "3.4.3"
}
```

But obviously I'd rather use the latest React version rather than having to bump it back to get the install to behave.

*alongside deleting package-lock.json and node_modules and re-running `npm install`

**Comment by james-work-account** [comment]
@bothwellw another workaround which seems to have temporarily fixed it is this:

- manually put the semantic dependencies in your package.json

```
"dependencies": {
  "react": "^17.0.1",
  "react-dom": "^17.0.1",
  "react-scripts": "4.0.1",
  "semantic-ui-css": "^2.4.1",
  "semantic-ui-react": "^2.0.1"
}
```

- Delete node_modules and package-lock.json
- Run `npm install --legacy-peer-deps`
- Restart react (stop the server, `npm start`)

I'm not happy with this as a solution as it still doesn't address the issue that something looks like it's relying on an old version of React, but it gets rid of the errors at least.

**Comment by brianespinosa** [comment]
You do need to have Semantic UI React in your dependencies. The shared package.json did not have it.

**Comment by james-work-account** [comment]
> You do need to have Semantic UI React in your dependencies. The shared package.json did not have it.

@brianespinosa That's because the `npm install ...` step failed, not because it isn't required. I don't know if yarn behaves differently but AFAIK if `npm install XYZ` fails, package.json isn't updated.

**Comment by xtrombone** [comment]
React 17 support is unreleased, correct?

**Comment by layershifter** [comment]
> React 17 support is unreleased, correct?

@xtrombone Yup, I will bump related packages tomorrow and we will release it ASAP. It should solve install issues with NPM 7.

**Comment by layershifter** [comment]
Folks, `semantic-ui-react@2.0.2` with #4138 was released yesterday. Now install passes without any warnings with NPM v7:

```
test-suir-v7> npm -v
7.3.0
```
```
test-suir-v7> npm install semantic-ui-react
added 13 packages, and audited 1713 packages in 6s
```

---

Thanks for reporting and @james-work-account for posting a workaround.

**Comment by asmtony** [comment]
This is also the case with React 18
36 silly fetch manifest react@^18.0.0
37 http fetch GET 200 https://registry.npmjs.org/react 41ms (cache revalidated)
38 silly fetch manifest react@^16.8.0 || ^17.0.0
39 timing idealTree Completed in 2214ms
40 timing command:install Completed in 2225ms
41 verbose stack Error: unable to resolve dependency tree

I am going to try the work arounds above

**Comment by alancphilpott** [comment]
> This is also the case with React 18 36 silly fetch manifest react@^18.0.0 37 http fetch GET 200 https://registry.npmjs.org/react 41ms (cache revalidated) 38 silly fetch manifest react@^16.8.0 || ^17.0.0 39 timing idealTree Completed in 2214ms 40 timing command:install Completed in 2225ms 41 verbose stack Error: unable to resolve dependency tree
> 
> I am going to try the work arounds above

I am also having the same issue with React 18 release. Did any work around fix the issue?

**Comment by asmtony** [comment]
No it did not. I had to revert back to React 17 so that Semantic would work. I got rid of the errors by doing the 
Delete node_modules and package-lock.json
Run npm install --legacy-peer-deps
Restart react (stop the server, npm start)

But this just stopped any styling being applied?

**Comment by galustee** [comment]
I ran into this as well. It's working on React 18 when I installed it like:

`npm install semantic-ui-react semantic-ui-css --force`

However importing the css like `import 'semantic-ui-css/semantic.min.css'` didn't work.
I had to add the following to my index.html

```
  <link
    async
    rel="stylesheet"
    href="https://cdn.jsdelivr.net/npm/semantic-ui@2/dist/semantic.min.css"
  />
```

**Comment by just-hope-netizen** [comment]
@galustee hey, did you workaround had any drawbacks?

**Comment by alancphilpott** [comment]
> I ran into this as well. It's working on React 18 when I installed it like:
> 
> `npm install semantic-ui-react semantic-ui-css --force`
> 
> However importing the css like `import 'semantic-ui-css/semantic.min.css'` didn't work. I had to add the following to my index.html
> 
> ```
>   <link
>     async
>     rel="stylesheet"
>     href="https://cdn.jsdelivr.net/npm/semantic-ui@2/dist/semantic.min.css"
>   />
> ```

> @galustee hey, did you workaround had any drawbacks?

Best to wait for an official Semantic update. Revert to React 17.0.2. Then again, depends on your other dependencies but in my case, deleting node_modules and then manually reverting the React version, then running npm i again then the trick.

**Comment by DeveloperAlly** [comment]
Don't know why this is closed. This still isn't resolved. Off to another lib *shrug

**Comment by layershifter** [comment]
This issue was related to React 17. Changes need to allow React 18 are in `master` and will be released soon

**Comment by antguider** [comment]
Still seeing this issue

**Comment by layershifter** [comment]
> This issue was related to React 17. Changes need to allow React 18 are in `master` and will be released soon

Released in `semantic-ui-react@2.1.3`.
